src/hooks/useActiveProject.tsx

```diff
@@ -68,97 +69,98 @@ export const ActiveProjectContextProvider = ({
 
   const useMeResult = useMe();
   const useMeLoading = useMeResult.isLoading || !useMeResult.isFetched;
-  const [hookReturnValue, setHookReturnValue] = useState<ActiveProjectProps>(
-    {},
-  );
   const { data: userData } = useUser();
   const userId = userData?.id;
+  const [selectedId, setSelectedId] = useState<string>();
   const [previousUserId, setPreviousUserId] = useState(userId);
-  const [isFetched, setIsFetched] = useState<boolean>(false);
   const [storedProjectIdResult, setStoredProjectId] = useAsyncStorage(
     `${selectedProjectIdKey}:${userId}`,
   );
 
   /**
    * Initial setting of activeProject
    */
-  useEffect(() => {
+  const hookReturnValue = useMemo<ActiveProjectProps>(() => {
     if (
       !userId || // wait for user id before reading and writing to storage
-      hookReturnValue.activeProject?.id || // active project already set
       projectLoading || // wait for projects endpoint
       useMeLoading || // wait for subjects endpoint
-      (storedProjectIdResult.isLoading && !storedProjectIdResult.isError) // wait for async storage result or error
+      storedProjectIdResult.isLoading ||
+      storedProjectIdResult.isError // wait for async storage result or error
     ) {
-      return;
+      return {};
     }
 
     const { selectedProject, selectedSubject } = findProjectAndSubjectById(
-      storedProjectIdResult.data,
+      selectedId ?? storedProjectIdResult.data,
       projectsResult.data,
       useMeResult.data,
     );
 
     if (selectedProject && selectedSubject) {
-      setHookReturnValue({
+      return {
         activeProject: selectedProject,
         activeSubjectId: selectedSubject.subjectId,
-      });
-      setStoredProjectId(selectedProject.id);
+      };
     }
-    setIsFetched(true);
+
+    return {};
   }, [
     storedProjectIdResult.data,
     storedProjectIdResult.isLoading,
     storedProjectIdResult.isError,
     projectsResult.data,
     useMeResult.data,
-    hookReturnValue.activeProject?.id,
-    setStoredProjectId,
     projectLoading,
     useMeLoading,
     userId,
+    selectedId,
   ]);
 
+  useEffect(() => {
+    if (hookReturnValue?.activeProject?.id) {
+      setStoredProjectId(hookReturnValue?.activeProject?.id);
+    }
+  }, [hookReturnValue, setStoredProjectId]);
```

Comment:
Same here as comment in useActiveAccount - might make this first dep `hookReturnValue?.activeProject?.id` instead.